hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -771,6 +771,9 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 		}
 		validReleaseImage := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1.ValidReleaseImage))
 		if validReleaseImage != nil && validReleaseImage.Status == metav1.ConditionFalse {
+			if validReleaseImage.Reason == hyperv1.InvalidImageReason {
```

Comment:
returning an err with an invalid image is not ideal, since in most cases if you introduced an invalid image, err and retry won't help. You'd want to return nil and requeue when the watcher detects a change to the HC.release.

For the scenario in the PR (secret does not exist) we should watch the secret referenced, however that might be overcomplicated for the benefit we get, so for now erroring is fine but only for this case.  When running the validReleaseImage function we should check if the err is not found and reflect that in the Reason, so then here we check against secretNotFoundReason and only error on that.

Reply:
The thing is, If I don't return an error, the reconciliation get blocked. Initially it was as you said, returning `nil` and that is what we want to avoid. This is why I return an error.

Reply:
you only want to return error for the case where the secret does not exist, so you need a semantic in a Reason for that case.
Any other scenario which involves changing the HC e.g. badReleaseImage should keep returning nil